monai/inferers/splitter.py

```diff
@@ -58,101 +59,56 @@ def __call__(self, inputs: Any) -> Iterable[tuple[torch.Tensor, Sequence[int]]]:
 
 
 class SlidingWindowSplitter(Splitter):
+    """Split the input into patches with sliding window strategy and a possible overlap.
+    It also allows to offset the starting position and filter the patches.
+
+    Args:
+        patch_size : the size of the patches to be generated.
+        offset: the amount of offset for the patches with respect to the original input.  Defaults to 0.
+        overlap: the amount of overlap between patches in each dimension [0, 1). Defaults to 0.0.
+        filter_fn: a callable to filter patches. It should accepts exactly two parameters (patch, location), and
+            return True for a patch to keep. Defaults to no filtering.
+        device: the device where the patches are generated. Defaults to the device of inputs.
+        pad_kwargs: arguments for `torch.nn.functional.pad`.
+            To pad the input images in order to capture the patches crossing the border of the image
+            (either when the offset is negative or the image is non-divisible by the patch_size),
+            "mode" need to be set, for instance {"mode": "constant"}.
+
+    Note:
+        If only one scaler value is provided for `patch_size`, `offset`, or `overlap`,
+            it will be broadcasted to all the spatial dimensions.
+    """
+
     def __init__(
         self,
         patch_size: Sequence[int] | int,
         offset: Sequence[int] | int = 0,
         overlap: Sequence[float] | float = 0.0,
         filter_fn: Callable | None = None,
         device: torch.device | str | None = None,
-        pad_mode: str | None = PytorchPadMode.CONSTANT,
```

Comment:
May I know why remove this arg? In this case the user must set the pad mode if the pad is needed, perhaps we can keep this default value here, and keep it consistent with what in the transforms.
https://github.com/Project-MONAI/MONAI/blob/f754928bb220bda15f059779a5c45ca04a05ad5a/monai/transforms/croppad/array.py#L134-L141

Reply:
All the padding information is now captured with `pad_kwargs`, so `pad_mod` can be set as `{"mode": "constant"}`:
https://github.com/Project-MONAI/MONAI/blob/4d51cd118bad233e8255d63f41328037fa7b5268/monai/inferers/splitter.py#L72-L75